File: TAREA 3/2_DADOS_2_ENTEROS_POSITIVOS/Permutacion.py
# permutacion.py

def permutacion(n: int, k: int) -> int:
    """
    Calcula la permutación P(n, k) = n! / (n-k)!
    mediante multiplicación descendente.
    """
    if n < 0 or k < 0:
        raise ValueError("n y k deben ser enteros positivos")
    
    if k > n:
        return 0

    resultado = 1
    for i in range(n, n - k, -1):  # n * (n-1) * ... * (n-k+1)
        resultado *= i

    return resultado


def main():
    print("CÁLCULO DE PERMUTACIÓN P(n,k)\n")

    n = int(input("Ingrese n (entero positivo): "))
    k = int(input("Ingrese k (entero positivo): "))

    resultado = permutacion(n, k)
    print(f"\nP({n}, {k}) = {resultado}")


if __name__ == "__main__":
    main()

File: TAREA 3/2_DADOS_2_ENTEROS_POSITIVOS/Permutacion.java
// Permutacion.java
import java.math.BigInteger;
import java.util.Scanner;

public class Permutacion {

    public static BigInteger permutacion(int n, int k) {
        if (n < 0 || k < 0)
            throw new IllegalArgumentException("n y k deben ser positivos");

        if (k > n)
            return BigInteger.ZERO;

        BigInteger result = BigInteger.ONE;

        for (int i = n; i > n - k; i--) {
            result = result.multiply(BigInteger.valueOf(i));
        }

        return result;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.println("CÁLCULO DE PERMUTACIÓN P(n,k)");
        System.out.print("Ingrese n: ");
        int n = sc.nextInt();

        System.out.print("Ingrese k: ");
        int k = sc.nextInt();

        BigInteger resultado = permutacion(n, k);

        System.out.println("\nP(" + n + ", " + k + ") = " + resultado);
        sc.close();
    }
}
